From d22e0a07ff7743274e7164749593d27d14972cd8 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Mon, 31 Mar 2025 14:01:51 +0800
Subject: [PATCH 181/236] mtk: hostapd: include BPCC in MLE STA control when
 handling ASSOC response

In MLE's STA control field, BPCC should be set only for the association
response.

To prevent adding more augments on the function
hostapd_eid_eht_basic_ml_common(), mld_info->mld_sta, which is
set to true when handling association request in hostapd, is used help
determine that if the current frame is an association response.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/ieee802_11_eht.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/ieee802_11_eht.c b/src/ap/ieee802_11_eht.c
index 0f29cea47..f6e3d1916 100644
--- a/src/ap/ieee802_11_eht.c
+++ b/src/ap/ieee802_11_eht.c
@@ -1018,7 +1018,7 @@ static size_t hostapd_eid_eht_ml_len(struct mld_info *info,
 	for (link_id = 0; info && link_id < ARRAY_SIZE(info->links);
 	     link_id++) {
 		struct mld_link_info *link;
-		size_t sta_len = include_mld_id ? 21 : 22;
+		size_t sta_len = EHT_ML_STA_INFO_LEN;
 
 		link = &info->links[link_id];
 		if (!link->valid)
-- 
2.45.2

